README.md

```diff
@@ -18,4 +18,129 @@ Also you have to configure the `github` maven repository. You can either include
     <id>github</id>
     <url>https://andrsuh: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@maven.pkg.github.com/andrsuh/tiny-event-sourcing</url>
 </repository>
-```
\ No newline at end of file
+```
+# Example of how to use library
+## Theory
+https://www.eventstore.com/event-sourcing - a good source of information about event sourcing and CQRS.
+## Example
+First, when we implement events sourcing pattern we have to define aggregates. In this example we will be having user aggregate.
+```kotlin
+@AggregateType(aggregateEventsTableName = "aggregate-user")
+data class UserAggregate(
+    override val aggregateId: String
+) : Aggregate {
+    override var createdAt: Long = System.currentTimeMillis()
+    override var updatedAt: Long = System.currentTimeMillis()
+
+    var userName: String = ""
+    var userLogin: String = ""
+    var userPassword: String = ""
+    lateinit var defaultPaymentId: UUID
+    lateinit var defaultAddressId: UUID
+    var paymentMethods = mutableMapOf<UUID, PaymentMethod>()
+    var deliveryAddresses = mutableMapOf<UUID, DeliveryAddress>()
+}
+```
+@AggregateType annotations makes library understand that this is the aggregate recognisable by library.
```

Comment:
+ extends Aggregate
+ meta-information in AggregateType

Reply:
I've added this to readme #

> * extends Aggregate
> 
>     * meta-information in AggregateType

@AggregateType annotations extends aggregate and stores MetaInformation about aggregate recognisable by library. Each aggregate has its own domain events. They're written in UserAggregateDomainEvents.kt. This is the example of one of the events:

Reply:
User written class should be extending aggregate not Annotation